brave-tracefilters/src/main/java/com/github/kristofa/brave/tracefilter/ZooKeeperSamplingTraceFilter.java

```diff
@@ -1,170 +0,0 @@
-package com.github.kristofa.brave.tracefilter;
-
-import java.io.Closeable;
-import java.io.IOException;
-import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicInteger;
-import java.util.logging.Level;
-import java.util.logging.Logger;
-
-import org.apache.curator.RetryPolicy;
-import org.apache.curator.framework.CuratorFramework;
-import org.apache.curator.framework.CuratorFrameworkFactory;
-import org.apache.curator.framework.state.ConnectionState;
-import org.apache.curator.framework.state.ConnectionStateListener;
-import org.apache.curator.retry.ExponentialBackoffRetry;
-import org.apache.zookeeper.WatchedEvent;
-import org.apache.zookeeper.Watcher;
-import org.apache.zookeeper.Watcher.Event.EventType;
-import org.apache.zookeeper.data.Stat;
-
-import com.github.kristofa.brave.TraceFilter;
-
-import static com.github.kristofa.brave.internal.Util.checkNotBlank;
-import static java.lang.String.format;
-
-/**
- * {@link TraceFilter} that gets sample rate from ZooKeeper. It watches a ZooKeeper znode which contains the sample rate. If
- * the value changes our TraceFilter is notified and adapts the sample rate accordingly.
- * <ul>
- * <li>If sample rate value <= 0 tracing is disabled. In that case we won't trace any request.</li>
- * <li>If sample rate value = 1 we will trace every request.</li>
- * <li>If sample rate value > 1, for example 5, we will trace every 5 requests.</li>
- * <li>If the znode does not exist, gets deleted or has no value tracing will be disabled.</li>
- * </ul>
- * 
- * @author kristof
- */
-public class ZooKeeperSamplingTraceFilter implements TraceFilter, Watcher, Closeable {
-
-    private final static Logger LOGGER = Logger.getLogger(ZooKeeperSamplingTraceFilter.class.getName());
-    private final static int DEFAULT_SAMPLE_RATE = 0;
-
-    private final CuratorFramework zkCurator;
-    private final CountDownLatch connectionEstablished = new CountDownLatch(1);
-    private final String sampleRateZNode;
-
-    private int sampleRate;
-    private final AtomicInteger counter = new AtomicInteger();
-
-    /**
-     * Creates a new instance. If the initial connection with ZooKeeper can't be established within 2 seconds an unchecked
-     * exception will be thrown as this will probably indicate wrong configuration.
-     * 
-     * @param connectionString ZooKeeper connection string. Should not be <code>null</code> or empty.
-     * @param sampleRateZNode The znode that contains sample rate. Should not be <code>null</code> or empty.
-     * @throws IOException In case we can't connect with ZooKeeper.
-     * @throws InterruptedException In case we can't connect with ZooKeeper.
-     */
-    public ZooKeeperSamplingTraceFilter(final String connectionString, final String sampleRateZNode)
```

Comment:
this is a lot of red, but actually the new class is almost identical. Just that instead of swapping sampleRate, it swaps an instance of traceSampler.